drivers/spi/spi_ll_stm32.h

```diff
@@ -219,23 +219,15 @@ static inline void ll_func_set_fifo_threshold_16bit(SPI_TypeDef *spi)
 static inline void ll_func_disable_spi(SPI_TypeDef *spi)
 {
 #if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_spi)
-	if (LL_SPI_IsActiveMasterTransfer(spi)) {
-		LL_SPI_SuspendMasterTransfer(spi);
-		while (LL_SPI_IsActiveMasterTransfer(spi)) {
-			/* NOP */
-		}
+	/* Flush RX buffer */
+	while (LL_SPI_IsActiveFlag_RXP(spi)) {
+		(void)LL_SPI_ReceiveData8(spi);
 	}
 
 	LL_SPI_Disable(spi);
 	while (LL_SPI_IsEnabled(spi)) {
 		/* NOP */
 	}
-
-	/* Flush RX buffer */
-	while (LL_SPI_IsActiveFlag_RXP(spi)) {
-		(void)LL_SPI_ReceiveData8(spi);
-	}
-	LL_SPI_ClearFlag_SUSP(spi);
 #else
 	LL_SPI_Disable(spi);
 #endif /* st_stm32h7_spi */
```

Comment:
Likely, this can then be simplified to:
```suggestion
#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_spi)
	/* Flush RX buffer */
	while (LL_SPI_IsActiveFlag_RXP(spi)) {
		(void)LL_SPI_ReceiveData8(spi);
	}
#endif /* st_stm32h7_spi */

	LL_SPI_Disable(spi);
	while (LL_SPI_IsEnabled(spi)) {
		/* NOP */
	}
```

Reply:
I agree it can be simplified that way, but that will affect non-H7 devices, which I cannot test. Are you happy with this? If so, I will go ahead and apply it.

Reply:
We'll test it, but I don't expect issues to be seen.